src/test/java/org/broadinstitute/hellbender/tools/CountReadsIntegrationTest.java

```diff
@@ -11,11 +11,11 @@
 public final class CountReadsIntegrationTest extends CommandLineProgramTest {
 
     @Test(dataProvider = "filenames")
-    public void testCountReads(final String fileIn, final String referenceName) throws Exception {
+    public void testCountReads(final String prefix, final String fileIn, final String referenceName) throws Exception {
         final File ORIG_BAM = new File(getTestDataDir(), fileIn);
         final ArgumentsBuilder args = new ArgumentsBuilder();
         args.add("--input");
-        args.add(ORIG_BAM.getAbsolutePath());
+        args.add(prefix + ORIG_BAM.getAbsolutePath());
```

Comment:
Instead of embedding this within the integration tests for `CountReads`, create a `TaggedArgumentIntegrationTest` that uses a private minimal `ReadWalker`, and within the walker access the read arg(s) by symbolic name and do an assert.